python/pyarrow/_compute.pyx

```diff
@@ -140,6 +162,29 @@ cdef class ScalarAggregateKernel(Kernel):
 
 
 cdef class Function(_Weakrefable):
+    """
+    A compute function.
+
+    A function implements a certain logical computation over a range of
+    possible input signatures.  Each signature accepts a range of input
+    types and is implemented by a given Kernel.
+
+    Functions can be of different kinds:
+
+    * "scalar" functions apply an item-wise computation over all items
+      of their inputs.  Each item in the output only depends on the values
+      of the inputs at the same position.  Examples: addition, comparisons,
+      string predicates...
+
+    * "vector" functions apply a collection-wise computation, such that
+      each item in the output may depend on the values of several items
+      in each input.  Examples: dictionary encoding, sorting, extracting
+      unique values...
+
+    * "aggregate" functions reduce the dimensionality of the inputs by
+      applying a reduction function.  Examples: sum, minmax, mode...
+
```

Comment:
Should "meta" be explained here as well?

Reply:
Yes, you're right.